src/services/signatureHelp.ts

```diff
@@ -375,23 +375,41 @@ namespace ts.SignatureHelp {
                 const typeParameters = (candidateSignature.target || candidateSignature).typeParameters;
                 signatureHelpParameters = typeParameters && typeParameters.length > 0 ? map(typeParameters, createSignatureHelpParameterForTypeParameter) : emptyArray;
                 suffixDisplayParts.push(punctuationPart(SyntaxKind.GreaterThanToken));
-                const parameterParts = mapToDisplayParts(writer =>
-                    typeChecker.getSymbolDisplayBuilder().buildDisplayForParametersAndDelimiters(candidateSignature.thisParameter, candidateSignature.parameters, writer, invocation));
+                const parameterParts = mapToDisplayParts(writer => {
+                    const printer = createPrinter({ removeComments: true });

```

Comment:
`Printer` resets after each use. Rather than reallocating a `printer` each time, I'd allocate `printer` once and reuse it.